File: spring-webflux/src/main/java/software/darkmatter/school/blog/domain/post/data/PostRepository.java
package software.darkmatter.school.blog.domain.post.data;

import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface PostRepository extends ReactiveCrudRepository<Post, Long> {

    Mono<Post> findByIdAndDeletedAtIsNull(Long id);

    Flux<Post> findAllByDeletedAtIsNull(Pageable pageable);
}

File: kotlin-spring-webflux/src/main/kotlin/software/darkmatter/school/blog/domain/post/data/PostRepository.kt
package software.darkmatter.school.blog.domain.post.data

import kotlinx.coroutines.flow.Flow
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.reactive.ReactiveCrudRepository

interface PostRepository : ReactiveCrudRepository<Post, Long> {

    suspend fun findByIdAndDeletedAtIsNull(id: Long): Post?

    fun findAllByDeletedAtIsNull(pageable: Pageable): Flow<Post>
}
